extensions/kafka/src/main/java/io/deephaven/kafka/ingest/KafkaStreamPublisher.java

```diff
@@ -203,7 +203,8 @@ public synchronized long consumeRecords(@NotNull final List<? extends ConsumerRe
                     : null;
 
             for (ConsumerRecord<?, ?> record : records) {
-                if (--remaining == 0) {
+                // Note we only flush if we are about to overflow the chunks.
+                if (--remaining < 0) {
```

Comment:
My preferred version looks like:
`remaining == 0`
Removed `-1` from the re-initialization of remaining after making new sink chunks
`--remaining` at the end of the for loop